src/cdomains/apron/linearTwoVarEqualityDomain.apron.ml

```diff
@@ -195,32 +276,34 @@ struct
     let open Apron.Texpr1 in
     let exception NotLinearExpr in
     let exception ScalarIsInfinity in
-    let negate coeff_var_list = List.map (fun (coeff, var) -> (Z.(-coeff), var)) coeff_var_list in
-    let multiply_with_Z number coeff_var_list =
-      List.map (fun (coeff, var) -> (Z.(number * coeff, var))) coeff_var_list in
+    let negate coeff_var_list = List.map (function
+        | (Some(coeff,i),offs,divi) -> (Some(Z.neg coeff,i),Z.neg offs,divi)
+        | (None         ,offs,divi) -> (None               ,Z.neg offs,divi)) coeff_var_list in
+    let multiply_with_Q dividend divisor coeff_var_list = List.map (function
+        | (Some (coeff, var),offs,divi) -> Rhs.canonicalize (Some(Z.mul dividend coeff,var),Z.(dividend * offs),Z.mul divi divisor)
+        | (None,offs,divi)              -> Rhs.canonicalize (None,Z.mul dividend offs,Z.mul divi divisor)) coeff_var_list in
```

Comment:
Maybe `BatOption.map`? The other components are the same in both cases but given differently.